src/app/campaign/form/campaign-form.component.ts

```diff
@@ -95,17 +96,21 @@ export class CampaignFormComponent implements OnInit {
   }
 
   updateCampaignForm({ name, type, status, repName, notes, set_account_uri, set_advertiser_uri }: Campaign) {
-    this.campaignForm.reset(
-      {
-        set_account_uri,
-        name,
-        type,
-        status,
-        repName,
-        notes,
-        set_advertiser_uri
-      },
-      { emitEvent: false }
-    );
+    // when user types an advertiser name match into the advertiser autoselect, set advertiser by value (URI)
```

Comment:
This method runs every keydown in the autocomplete, so doing the campaignForm.reset on all fields for every update was causing the advertiser field to clear out. I moved over to this `patchValue` with (very important) options `{emitEvent: false, onlySelf: true}`  and conditionally only included fields that were present.

When typed in, the advertiser value is a "name" but when selected from the autocomplete dropdown, the value is the URI, so this sets the advertiser whether they typed the full matching value (without selecting) or selected one. (There's maybe some further weird UX with autocomplete, I'm sure we'll talk about it more.)